pkg/controller/pcidevice/pcidevice_controller.go

```diff
@@ -25,20 +30,30 @@ const (
 type Handler struct {
 	client        ctl.PCIDeviceClient
 	pci           *ghw.PCIInfo
+	nodeCache     ctlcorev1.NodeCache
+	networkCache  ctlnetworkv1beta1.VlanConfigCache
 	skipAddresses []string
 }
 
 func Register(
 	ctx context.Context,
 	pd ctl.PCIDeviceClient,
-) error {
+	coreFactory *ctlcore.Factory,
+	networkFactory *ctlnetwork.Factory) error {
 	logrus.Info("Registering PCI Devices controller")
 
 	handler := &Handler{
-		client: pd,
+		client:       pd,
+		nodeCache:    coreFactory.Core().V1().Node().Cache(),
+		networkCache: networkFactory.Network().V1beta1().VlanConfig().Cache(),
 	}
 
 	nodename := os.Getenv("NODE_NAME")
+
+	if err := start.All(ctx, 1, coreFactory, networkFactory); err != nil {
```

Comment:
can we do this in https://github.com/harvester/pcidevices/blob/0a1f7c2b535e5729aa05dcdb47a28b5a0cafb83b/main.go#L123?

Reply:
the pcidevice register is a blocking call, hence the need to run it within  the register method.